src/apps/lwaftr/lwaftr.lua

```diff
@@ -531,32 +531,40 @@ local function get_ipv6_dst_ip(lwstate, pkt)
    return fstring(pkt.data + ipv6_dst, 16)
 end
 
-local function cache_ipv6_fragment(lwstate, frag)
-   local cache = lwstate.fragment6_cache
+local function key_ipv6_frag(lwstate, frag)
    local frag_id = fragmentv6.get_ipv6_frag_id(frag, lwstate.l2_size)
    local src_ip = get_ipv6_src_ip(lwstate, frag)
    local dst_ip = get_ipv6_dst_ip(lwstate, frag)
    local src_dst = src_ip..dst_ip
-   cache[frag_id] = cache[frag_id] or {}
-   cache[frag_id][src_dst] = cache[frag_id][src_dst] or {}
-   table.insert(cache[frag_id][src_dst], frag)
-   return cache[frag_id][src_dst], frag_id
+   return frag_id .. '|' .. src_dst
 end
 
-local function clean_fragment_cache(lwstate, frag_id)
-   lwstate.fragment6_cache[frag_id] = nil
+local function cache_ipv6_fragment(lwstate, frag)
+   local cache = lwstate.fragment6_cache
+   local key = key_ipv6_frag(lwstate, frag)
+   cache[key] = cache[key] or {}
+   table.insert(cache[key], frag)
+   return cache[key]
+end
+
+local function clean_ipv6_fragment_cache(lwstate, frags)
+   local key = key_ipv6_frag(lwstate, frags[1])
+   lwstate.fragment6_cache[key] = nil
+   for i=1,#frags do
+      packet.free(frags[i])
+   end
```

Comment:
This @eugeneia comment in a recent PR came to my mind https://github.com/SnabbCo/snabbswitch/pull/664

" Do not keep invalid pointers around. If you free a packet held in a variable that can be accessed outside the lexical scope (for instance an object slot), null it right away."

Maybe it's not necessary here, but what do you think about nulling the packet as well?

```lua
packet.free(frags[i])
frags[i] = nil
```